compiler/noirc_frontend/src/elaborator/statements.rs

```diff
@@ -432,9 +432,26 @@ impl<'context> Elaborator<'context> {
     fn elaborate_comptime_statement(&mut self, statement: Statement) -> (HirStatement, Type) {
         let span = statement.span;
         let (hir_statement, _typ) = self.elaborate_statement(statement);
-        let mut interpreter = Interpreter::new(self.interner, &mut self.comptime_scopes);
+        let mut interpreter_errors = vec![];
+        let mut interpreter = Interpreter::new(
+            self.interner,
+            &mut self.comptime_scopes,
+            self.debug_comptime_scope,
+            &mut interpreter_errors,
+        );
```

Comment:
Repeated code here (between here & elaborate comptime block) with setting up the interpreter, running a function, then checking debug_comptime_scope.

Reply:
I was able to simplify a bunch of it. For the remainder of the repeated code, there are enough differences and mutable borrows that I'd expect a common function to be more clunky.

What do you think?

Reply:
Looks a lot better